### PR TITLE
Use `cupy.core._dtype.to_cuda_dtype` whenever possible

### PR DESCRIPTION
**Pull request by leofang**
Reduce a bit of code duplication.

TODO:
- [x] Check if CPU overhead is reduced.

### CODE REVIEW

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
CPU overhead remains the same.

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by asi1024** [comment]
@leofang Could you resolve conflicts?

**Comment by leofang** [comment]
@asi1024 Thanks for reminder, I just did.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Review by asi1024 — APPROVED** [review]

File: cupy/cusparse.py [hunk 1]
```diff
@@ -73,19 +73,6 @@ def _call_cusparse(name, dtype, *args):
     return f(*args)
 
 
-def _dtype_to_DataType(dtype):
```
Comment (asi1024) [inline cupy/cusparse.py:76]: This function is called also from `cupyx/scipy/sparse/linalg/_eigen.py` and `cupyx/scipy/sparse/linalg/_iterative.py`. Could you fix them?

Comment (leofang) [reply cupy/cusparse.py:76]: Thanks, @asi1024! Should be fixed in e4bede5.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM! Thanks!

**Comment by leofang** [comment]
Thanks 🙏

**Pull request merged**